controllers/resource/reconciler.go

```diff
@@ -164,6 +171,16 @@ func (cor *clusterReconciler) Reconcile(ctx context.Context, objectKey types.Nam
 	return cor.applyTemplates(ctx, cs, resources, dryRun)
 }
 
+func (cor *clusterReconciler) setBundlesRef(ctx context.Context, cluster *anywherev1.Cluster) error {
+	mgmtClusterKey := types.NamespacedName{Name: cluster.ManagedBy(), Namespace: cluster.Namespace}
+	mgmtCluster, err := cor.FetchCluster(ctx, mgmtClusterKey)
+	if err != nil {
+		return err
+	}
+	cluster.Spec.BundlesRef = mgmtCluster.Spec.BundlesRef
```

Comment:
for a self-managed cluster, this will just set BundlesRef to itself, won't it? So if it was nil before, it'll be nil still.